docs/adr/2019-04-23-transcript-architecture.md

```diff
@@ -202,5 +206,32 @@ test integration locally with Lambdas and Gateways.
 
 #### Disadvantages
 
-* There could be a timeout issue for Lambda
+* There could be a time-out issue for Lambda
 * Difficulty in debugging due to Lambda
+* Operating System is abstracted away (loss of control)
+
+### Option 3
+
+This has been developed two months after. The architecture is fleshed out in this option.
+It shows, clearer responsibilities with defined interfaces.
+We've added:
+
+1. SNS and SQS (fanout pattern) for reliable job delivery to microservices.
+2. S3 signed URL communication via API, rather than direct upload from API. This will improve data transmission from different components.
+3. PSTT Queue jobs to be published by Audio FFMPEG Service, subscribed PSTT client, and pushed to PSTT.
+4. On completion of PSTT task, a notification will be pushed to the PSTT Queue, which the client will consume to update the API.
```

Comment:
In this ADR there should be no mention of the Platform STT service.

The logic to connect to platform STT service should not bleed into the rest of our system.

The system uses the [`STT-Proxy`](https://github.com/bbc/digital-paper-edit-stt-proxy) service which provides an abstraction over any Speech to text provider we might be using (in this case PSTT) by providing
- A consistent interface (RESTfull API?) 
-  Separation of concern between between the inner working of a STT provider and our own system (PSTT has already announced that there will be breaking changes)
- Under the hood `STT-Proxy` implements an node SDK module to make and retrieve requests from PSTT.
- Data conversion between PSTT data model (similar but different from AWS STT) to `digital-paper-edit` transcript data schema.

Reply:
👍 thanks for reminding, I'll remove the very PSTT specific stuff and replace it with STT proxy.